api/src/main/java/org/pmiops/workbench/elasticsearch/ElasticSearchService.java

```diff
@@ -0,0 +1,55 @@
+package org.pmiops.workbench.elasticsearch;
+
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.elasticsearch.client.core.CountRequest;
+import org.elasticsearch.index.query.AbstractQueryBuilder;
+import org.elasticsearch.index.query.BoolQueryBuilder;
+import org.elasticsearch.index.query.TermQueryBuilder;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+
+@Service
+public class ElasticSearchService {
+
+  private static final String INDEX = "index";
+  private static final String TYPE = "type";
+  private RestHighLevelClient client;
+
+  @Autowired
+  public ElasticSearchService(RestHighLevelClient client) {
+    this.client = client;
+  }
+
+  /**
+   * This method will be used to generate counts from a
+   * org.pmiops.workbench.model.SearchRequest.
+   *
+   * @param userRequest
+   * @return
+   */
+  public Long elasticCount(org.pmiops.workbench.model.SearchRequest userRequest) throws IOException {
+    CountRequest countRequest = new CountRequest()
+      //What indices should we use here?
```

Comment:
These are in the design doc, "type" is pretty much deprecated and will be "_doc" for all types. The index will change depending on the CDR version.

Open question: do we want to leverage the CdrVersionContext here as well?

Reply:
I think we do.